mingw-w64-curl/PKGBUILD

```
@@ -126,10 +118,59 @@ build() {
   make
 }
 
-package() {
-  cd "${srcdir}/build-${CARCH}"
+build() {
+  do_build -openssl
+  do_build -winssl
+  if [[ ${MINGW_PACKAGE_PREFIX} != *-clang-* ]]; then
+    do_build -gnutls
+  fi
+}
+
+do_package() {
+  cd "${srcdir}/build-${CARCH}$1"
   make DESTDIR="${pkgdir}" install
 
   local PREFIX_DEPS=$(cygpath -am ${MINGW_PREFIX})
   sed -s "s|${PREFIX_DEPS}|${MINGW_PREFIX}|g" -i ${pkgdir}${MINGW_PREFIX}/bin/curl-config
 }
+
+package_curl() {
+  depends=("${depends[@]}"
+           "${_cert_depends[@]}"
+           "${_openssl_depends[@]}")
+  conflicts=("${MINGW_PACKAGE_PREFIX}-${_realname}-winssl"
+             "${MINGW_PACKAGE_PREFIX}-${_realname}-gnutls")
+  do_package
+}
+
+package_curl-winssl() {
+  depends=("${depends[@]}"
+           "${MINGW_PACKAGE_PREFIX}-libssh2-wincng")
+  provides=("${MINGW_PACKAGE_PREFIX}-${_realname}")
+  conflicts=("${MINGW_PACKAGE_PREFIX}-${_realname}"
+             "${MINGW_PACKAGE_PREFIX}-${_realname}-gnutls")
+  do_package -winssl
+}
+
+package_curl-gnutls() {
+  provides=("${MINGW_PACKAGE_PREFIX}-${_realname}")
+  depends=("${depends[@]}"
+           "${_cert_depends[@]}"
+           "${_gnutls_depends[@]}")
+  conflicts=("${MINGW_PACKAGE_PREFIX}-${_realname}"
+             "${MINGW_PACKAGE_PREFIX}-${_realname}-winssl")
+  if [[ ${MINGW_PACKAGE_PREFIX} != *-clang-* ]]; then
+    do_package -gnutls
+  fi
```

This one wasn't necessary before, as the package function would not be called if the corresponding package wasn't listed in `pkgname`
```suggestion
  do_package -gnutls
```